contentcuration/contentcuration/frontend/channelEdit/views/progress/ProgressModal.vue

```diff
@@ -1,82 +1,60 @@
-                {{ descriptionText }}
-              </p>
-              <ProgressBar
-                v-if="!nothingToSync"
-                :progressPercent="progressPercent"
-                :currentTaskError="currentTaskError"
-              />
-              <VLayout
-                v-if="currentTaskError"
-                row
-                class="caption mt-2 red--text"
-                data-test="error"
-              >
-                <VFlex class="pr-2">
-                  <Icon small color="red">
-                    error
-                  </Icon>
-                </VFlex>
-                <VFlex>{{ $tr('defaultErrorText') }}</VFlex>
-              </VLayout>
-            </VCardText>
+      {{ progressModalDescription }}
 
-            <VCardActions>
-              <VSpacer />
-              <VBtn
-                v-if="progressPercent === 100 || currentTaskError"
-                color="primary"
-                data-test="refresh"
-                @click="closeOverlay"
-              >
-                {{ doneButtonText || $tr('refreshButton') }}
-              </VBtn>
-              <VBtn v-else color="primary" data-test="stop" @click="step++">
-                {{ stopButtonText || $tr('stopButton') }}
-              </VBtn>
-            </VCardActions>
-          </VWindowItem>
+      <ProgressBar
+        v-if="!nothingToSync"
+        data-test="progress-bar"
+        :progressPercent="progressPercent"
+        :currentTaskError="currentTaskError"
+      />
+      <div
+        v-if="currentTaskError"
+        class="caption mt-2 red--text"
+      >
+        <Icon small color="red">
+          error
+        </Icon>
+        {{ $tr('defaultErrorText') }}
+      </div>
 
-          <VWindowItem :value="2">
-            <VCardTitle class="font-weight-bold pb-0 title">
-              {{ cancelHeaderText || $tr('cancelHeader') }}
-            </VCardTitle>
-            <VCardText class="py-4">
-              {{ cancelText || $tr('cancelText') }}
-            </VCardText>
-
-            <VCardActions>
-              <VSpacer />
-              <VBtn flat data-test="cancelstop" @click="step--">
-                {{ $tr('cancel') }}
-              </VBtn>
-              <VBtn color="primary" data-test="confirmstop" @click="cancelTask">
-                {{ stopButtonText || $tr('confirmStopButton') }}
-              </VBtn>
-            </VCardActions>
-          </VWindowItem>
-        </VWindow>
-      </VCard>
-    </VDialog>
+      <template slot="actions">
+        <KButton
+          v-if="isFinished || currentTaskError"
+          primary
+          data-test="refresh-button"
+          @click="closeOverlay"
+        >
+          {{ $tr('refreshButton') }}
+        </KButton>
+        <KButton
+          v-else
+          primary
+          data-test="stop-button"
+          @click="displayCancelModal = true"
+        >
+          {{ $tr('stopButton') }}
+        </KButton>
+      </template>
```

Comment:
We should favor using the props for the buttons (IMO) unless there is a stylistic need not to because that will guarantee the best consistency in styling.

If the only reason you've opted to use action slot here is because you need those `data-test` attrs then I think that's a code smell that we should address (although not here probably). 

If we need such granular tests that we are testing that a button exists - then that test should live in KDS on the KModal component because it's the component making sure that the button exists. We should instead only write tests for the things we control - which is the value of the props (or actios) that we pass into KModal. 

I wonder, if you do this through props and then mount `KModal` and trigger `submit` or `cancel` events on it I wonder if you can then just test that the code you've passed to the respective props is executed?

I can't say that I know the best answer or solution - but I definitely think we need to bear in mind how easy it is to use KDS components in tests as we continue working toward writing documentation for our components.

Reply:
Reviewing this - I realize now that the buttons are v-if/v-else - so my preceding comment is less relevant here than I thought at first.

Reply:
> If the only reason you've opted to use action slot here is because you need those data-test attrs then I think that's a code smell that we should address (although not here probably).

> Reviewing this - I realize now that the buttons are v-if/v-else - so my preceding comment is less relevant here than I thought at first.

Yes, the slot is there because of `v-if`s.

> If we need such granular tests that we are testing that a button exists - then that test should live in KDS on the KModal component because it's the component making sure that the button exists.

This is not the purpose of this test suite. It rather tests that when you click the button, a corresponding action will be triggered which has nothing to do with KModal itself. If I test for the presence of buttons, it's just to ensure that I can click them or they are buttons that are conditionally rendered, and the decision is a responsibility that belongs to ProgressModal.

> I wonder, if you do this through props and then mount KModal and trigger submit or cancel events on it I wonder if you can then just test that the code you've passed to the respective props is executed?

The following comment is a more general comment and I am not sure if it's relevant to what you meant by this as it looks to be related to a specific issue that you've described. As you mentioned on Slack, this can serve as a discussion for our further work.

I would have two options how to test this example scenario: "When I click confirm stop button on the cancel modal, a certain action will be triggered (e.g. stop publishing)"
(1) Find KModal and programmatically trigger 'submit'
(2) Find a cancel button and trigger 'click'

I prefer (2) because that's what the user is doing (I've tried to explain[ this a bit in this comment](https://github.com/learningequality/studio/pull/2968#discussion_r579381843)) and if we change some internals, e.g. swap the modal component for something else or even just rename some of its props or events, the test won't break. I would not expect tests to break as long as the user-facing interface is the same. In this PR, going with (1) wouldn't be such a big deal, though again, generally speaking, this approach very often causes extra maintenance work and sometimes gives bad results. Also, I understand the test suite to be a documentation of the component, and if its description says "When I click confirm stop button", that's what I would expect the test to do.

That said, in some cases, reaching into internals may make sense or may even be unavoidable. There is no perfect solution. We can chat about this more.

> but I definitely think we need to bear in mind how easy it is to use KDS components in tests as we continue working toward writing documentation for our components.
 
Happy to chat about it too. In this PR, I haven't had any problems with KModal, I could easily reach for a button [in this way](https://github.com/learningequality/studio/pull/2968/files#diff-a582b5d32b4621a4b891215c90825a3b18b5cde28fb2dce4742a41f0e15a6fd5R40). Even if the name attribute wasn't present, I could just find a second button, use a class, etc.